libafl/src/stages/logics.rs

```diff
@@ -8,28 +8,6 @@ use crate::{
     Error,
 };
 
-/// Progress for nested stages. This merely enters/exits the inner stage's scope.
```

Comment:
@addisoncrump @domenukk 

Can you check this part?
I think these state.enter_inner_stage() is only needed for IfElseStage where the stages bifurcate
and for IfElseStage, we do have these enter_inner_stage and exit_innter_stage()

Hence we don't need to add this in addition.

The current state, is calling enter_inner_stage() twice in IfElseStage which looks very odd